grass/kernel.c

```diff
@@ -61,16 +71,32 @@ void intr_entry(uint id) {
     if (earth->tty_recv_intr() && curr_pid >= GPID_USER_START) {
         /* User process killed by ctrl+c interrupt */
         INFO("process %d killed by interrupt", curr_pid);
-        asm("csrw mepc, %0" ::"r"(0x800500C));
+        proc_set[proc_curr_idx].mepc = (uint)sys_exit;
         return;
     }
 
     /* Ignore other interrupts for now */
     if (id == INTR_ID_SOFT) proc_syscall();
-    if (id == INTR_ID_TIMER) proc_yield();
+    proc_yield();
+}
+
+static void proc_request()
```

Comment:
I think we don't need this new function. See my comments below in `proc_yield`, but let me know if I am wrong.

Reply:
Updated `proc_yield` so it runs pending system calls, but using `else if` would lead to situations where no processes can be scheduled, but their pending system call would succeed and make them runnable.

Let me know how you feel about using two separate `if` statements, where we use a pointer to the process' PCB to allow for the process' status to be updated by the pending system call.